msteams-platform/bots/how-to/teams-conversational-ai/teams-conversation-ai-overview.md

```diff
@@ -141,21 +141,177 @@ The following table lists the updates to the Teams AI library:
+    {
+        "schema": 1.1,
+        "description": "",
+        "type": "",
+        "completion": {
+    +       "tool_choice": "auto",
+    +       "parallel_tool_calls": true,
+        },
+    +    "augmentation": {
+    +        "augmentation_type": "tools"
+    +    }
+    }
+    ```
+
+1. Specify all your `function definitions` in the `actions.json` file, which is in the `prompts` folder. Ensure that you follow the schema to avoid errors when the action is called by the LLM.
+
+    ```JSON
+    [{
+        "name": "CreateList",
+        "description": "Creates a list"
+    }]
+    ```
+
+1. Register your `handlers` in your `application` class.
+
+    * Each handler is a callback function that runs when a specific event happens. The function call handler executes code in response to the event.
+    * The function call must return a string as the output of the function call.
+    * After all functions are executed, the outputs are returned to the model. When the model requests to invoke any functions, these are mapped to `DO` commands within a `Plan` and are invoked in the AI class `run` function. The outputs are then returned to the model with tool call IDs to show that the tools were used.
+
+   The following code snippet shows how to register `handlers`:
+
+    # [JavaScript](#tab/javascript1)
+
+    ```JavaScript
+    app.ai.action("createList", async (context: TurnContext, state: ApplicationTurnState, parameters: ListAndItems) => {
+    // Ex. create a list with name "Grocery Shopping".
+    ensureListExists(state, parameters.list);
+    return `list created and items added. think about your next action`;
+    });
+    ```
+
+    # [C#](#tab/dotnet1)
+
+    ```C#
+    [Action("CreateList")]
+    public string CreateList([ActionTurnState] ListState turnState, [ActionParameters] Dictionary<string, object> parameters)
+    {
+        ArgumentNullException.ThrowIfNull(turnState);
+        ArgumentNullException.ThrowIfNull(parameters);
+
+        string listName = GetParameterString(parameters, "list");
+
+        EnsureListExists(turnState, listName);
+
+        return "list created. think about your next action";
+    }
+    ```
+
+    # [Python](#tab/python1)
+
+    ```Python
+    @app.ai.action("createList")
+    async def create_list(context: ActionTurnContext, state: AppTurnState):
+    ensure_list_exists(state, context.data["list"])
+    # Continues exectuion of next command in the plan.
+    return ""
+    ```
+
+    ---
+
+### Enable tool options
+
+You can enable the following tool options:
+
+* **Enable Tool Choice**: To allow the model to select the function it must call by enabling tool selection. In the `config.json` file:
+
+    - Set `tool_choice` as `required` to mandate the model to always call at least one function. 
+    - Set `tool_choice` to a specific function using its definition for using that function. 
+    - Set  `tool_choice` as `none` to disable tool choice.
+
+  Default value of `tool_choice` is `auto`.
```

Comment:
add "This means the model can select which functions to call"

Reply:
@lilyydu The section already says this in the bullet point:

 **Enable Tool Choice**: To allow the model to select the function it must call by enabling tool selection.

Reply:
suggested this because I don't think it's intuitive for the developer to know that "auto" means the model gets to choose whichever functions it wants to call

Reply:
@lilyydu Thank you for the input. Updated as suggested.